packages/xstate-react/test/types.test.tsx

```diff
@@ -57,3 +64,52 @@ describe('useService', () => {
     render(<Todo index={0} />);
   });
 });
+
+describe('useMachine', () => {
+  it('should preserve typestate information.', () => {
```

Comment:
I'm wondering - so far you have only changed `xstate`'s source files, but you have added a test to `@xstate/react`. Any particular reason behind this? Should we add similar tests to `xstate`? Were your changes essential for `@xstate/react`?

Reply:
The goal of the changes where to preserve typestate for components using `useMachine`.

Reply:
We should. I'm working on some additional typestate related tests for core

Reply:
> The goal of the changes where to preserve typestate for components using useMachine.

This was probably already fixed by [this commit](https://github.com/davidkpiano/xstate/commit/30c24bcc7fa7ea7127f3088fc69589d859044852#diff-dd3974b63e6fa683880195c4603244bb). Notice how the return type of `useMachine` is already typed with type states in mind, so your changes had no effect on it.

It's great to have tests to cover this now though!

Reply:
👍 I've added tests for coe